solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp2SolrClient.java

```diff
@@ -233,6 +256,10 @@ public Builder(List<String> zkHosts, Optional<String> zkChroot) {
       if (zkChroot.isPresent()) this.zkChroot = zkChroot.get();
     }
 
+    public Builder(ClusterStateProvider stateProvider) {
```

Comment:
It should have javadoc; in this case saying this is for an expert use-case (needn't say more).